### PR TITLE
lara/hang: fix swing-in on 1-click ledges

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TRX/blob/develop/docs/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [x] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

This adjusts the conditions for Lara's swing-in animation further, to prevent the animation from occurring on more invalid cases.

#### Testing

- [x] Check this out: [test-swing-in.txt](https://github.com/user-attachments/files/25413392/test-swing-in.txt)
- [x] Test more edge cases (pun intended)

### CODE REVIEW

**Review by aredfan — APPROVED** [review]

File: src/trx/game/lara/col/jump.c [hunk 1]
```diff
@@ -122,7 +137,6 @@ static bool M_TestHangJump(ITEM *const item, COLL_INFO *const coll)
     item->current_anim_state = anim->current_anim_state;
     item->goal_anim_state = anim->current_anim_state;
 
-    const BOUNDS_16 *const bounds = Item_GetBoundsAccurate(item);
```
Comment (lahm86) [inline src/trx/game/lara/col/jump.c:125]: This re-introduces #3132.  The bounds need updating after having switched animations.

Comment (rr-) [reply src/trx/game/lara/col/jump.c:125]: This is a terrible design!
Some would even say… Core Design.

Comment (rr-) [reply src/trx/game/lara/col/jump.c:125]: Fixed~

**Review by lahm86 — APPROVED** [review]
